vendor/k8s.io/kubernetes/openshift-kube-apiserver/admission/customresourcevalidation/apiserver/validate_apiserver.go

```diff
@@ -0,0 +1,145 @@
+package apiserver
+
+import (
+	"fmt"
+	"strings"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+
+	"k8s.io/apimachinery/pkg/api/validation"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/openshift-kube-apiserver/admission/customresourcevalidation"
+)
+
+func toAPIServerV1(uncastObj runtime.Object) (*configv1.APIServer, field.ErrorList) {
+	if uncastObj == nil {
+		return nil, nil
+	}
+
+	errs := field.ErrorList{}
+
+	obj, ok := uncastObj.(*configv1.APIServer)
+	if !ok {
+		return nil, append(errs,
+			field.NotSupported(field.NewPath("kind"), fmt.Sprintf("%T", uncastObj), []string{"APIServer"}),
+			field.NotSupported(field.NewPath("apiVersion"), fmt.Sprintf("%T", uncastObj), []string{"config.openshift.io/v1"}))
+	}
+
+	return obj, nil
+}
+
+type apiserverV1 struct {
+	infrastructureGetter func() configv1client.InfrastructuresGetter
+}
+
+func (a apiserverV1) ValidateCreate(uncastObj runtime.Object) field.ErrorList {
+	obj, errs := toAPIServerV1(uncastObj)
+	if len(errs) > 0 {
+		return errs
+	}
+
+	errs = append(errs, validation.ValidateObjectMeta(&obj.ObjectMeta, false, customresourcevalidation.RequireNameCluster, field.NewPath("metadata"))...)
+	errs = append(errs, validateAPIServerSpecCreate(obj.Spec)...)
+	errs = append(errs, a.validateSNINames(obj)...)
+
+	return errs
+}
+
+func (a apiserverV1) validateSNINames(obj *configv1.APIServer) field.ErrorList {
+	errs := field.ErrorList{}
+	if len(obj.Spec.ServingCerts.NamedCertificates) == 0 {
+		return errs
+	}
+
+	infrastructure, err := a.infrastructureGetter().Infrastructures().Get("cluster", metav1.GetOptions{})
+	if err != nil {
+		errs = append(errs, field.InternalError(field.NewPath("metadata"), err))
+	}
+	for i, currSNI := range obj.Spec.ServingCerts.NamedCertificates {
+		// if names are specified, confirm they do not match
+		// if names are not specified, the cert can still match, but only the operator resolves the secrets down.  We gain a lot of benefit by being sure
+		// we don't allow an explicit override of these values
+		for j, currName := range currSNI.Names {
+			path := field.NewPath("spec").Child("servingCerts").Index(i).Child("names").Index(j)
+			if currName == infrastructure.Status.APIServerInternalURL {
+				errs = append(errs, field.Invalid(path, currName, fmt.Sprintf("may not match internal loadbalancer: %q", infrastructure.Status.APIServerInternalURL)))
+				continue
+			}
+			if strings.HasSuffix(currName, ".*") {
```

Comment:
how do we match SNI names? longest win?

Reply:
> how do we match SNI names? longest win?

Yes, but right now (4.1.z), we don't specify any names, so anything with a wildcard wins.  We can relax this in 4.2 later if we have a reason, but this is the safest starting point.

Reply:
ic. make sense

Reply:
Do we have a bugzilla or jira to add the name in the operator generated cert? What about the ones from installer?

Reply:
> Do we have a bugzilla or jira to add the name in the operator generated cert? What about the ones from installer?

Do we need more than https://bugzilla.redhat.com/show_bug.cgi?id=1724189 ?